### PR TITLE
[SPARK-9695] [ML] Add random seed Param to ML Pipeline

### PR DESCRIPTION
**Pull request by yanboliang**
Add random seed Param to ML Pipeline

### CODE REVIEW

**Comment by yanboliang** [comment]
Jenkins, test this please.

File: mllib/src/main/scala/org/apache/spark/ml/Pipeline.scala [hunk 1]
```diff
@@ -100,6 +101,15 @@ class Pipeline(override val uid: String) extends Estimator[PipelineModel] {
   /** @group getParam */
   def getStages: Array[PipelineStage] = $(stages).clone()
 
+  /** @group setParam */
```
Comment (holdenk) [inline mllib/src/main/scala/org/apache/spark/ml/Pipeline.scala]: Probably worth it to include some Javadoc which explains how setSeed works (e.g. it seems to set the seed on all elements which haveSeed but are not set but a user might assume it overwrites the previous values).

File: mllib/src/test/scala/org/apache/spark/ml/PipelineSuite.scala [hunk 2]
```diff
@@ -25,6 +25,8 @@ import org.scalatest.mock.MockitoSugar.mock
 
 import org.apache.spark.SparkFunSuite
 import org.apache.spark.ml.feature.HashingTF
```
Comment (holdenk) [inline mllib/src/test/scala/org/apache/spark/ml/PipelineSuite.scala]: We could probably combine these three imports to a single line.

**Comment by holdenk** [comment]
So this is a bit out of sync with master, how would feel about updating it? Maybe @jkbradley as the reporter of this JIRA could also take a look once that is done.

**Comment by holdenk** [comment]
So also following up from @jkbradley's note on https://issues.apache.org/jira/browse/SPARK-7653 we don't currently expose HasSeed to the public so external models with seeds won't be able to implement it (using our public APIs). I think if we do go with this approach the most reasonable thing to do is to make HasSeed public but I'd love to hear others' thoughts on the matter.

**Pull request closed without merging**